mindmeld/augmentation.py

```diff
@@ -0,0 +1,285 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the data augmentation processes for MindMeld."""
+
+import logging
+import re
+
+from abc import ABC, abstractmethod
+from tqdm import tqdm
+
+from .components._config import get_augmentation_config, get_language_config
+from .constants import _get_pattern
+from .models.helpers import register_augmentor
+from .resource_loader import ResourceLoader
+
+logger = logging.getLogger(__name__)
+
+try:
+    import torch
+except ImportError:
+    logger.info(
+        "Library not found: 'torch'. Run 'pip install mindmeld[augment]'" " to install."
+    )
+
+try:
+    from transformers import PegasusForConditionalGeneration, PegasusTokenizer
+    from transformers import MarianMTModel, MarianTokenizer
+except ImportError:
+    logger.info(
+        "Library not found: 'transformers'. Run 'pip install mindmeld[augment]'"
+        " to install."
+    )
+
+# pylint: disable=R0201
+
+
+class Augmentor(ABC):
+    """
+    Abstract Augmentor class.
+    """
+
+    def __init__(self, app_path, config=None):
+        """Initializes an augmentor.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        self.app_path = app_path
+        self.config = config or get_augmentation_config(app_path=app_path)
+        self._resource_loader = ResourceLoader.create_resource_loader(app_path)
+
+    def augment(self, config, **kwargs):
+        """Augments queries given initial queries in application.
+
+        Args:
+            config (dict, optional): App config to use instead of class config.
+        """
+        config = config or self.config
+        filtered_paths = self._get_files(config)
+
+        for path in tqdm(filtered_paths):
+            queries = self._read_path_queries(path)
+            augmented_queries = self.augment_queries(queries, **kwargs)
+            self._write_files(path, augmented_queries)
+
+    @abstractmethod
+    def augment_queries(self, queries):
+        """Generates augmented data given application queries.
+
+        Args:
+            queries (list): List of queries.
+
+        Return:
+            augmented_queries (list): List of augmented queries.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    def _get_files(self, config):
```

Comment:
Pass in explicit paths

Reply:
This function takes in the config and resolves the rules set for paths

Reply:
I think its better for readability if paths was just pass in. The config is not being used for anything else here.